spring-kafka/src/test/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryServiceTests.java

```diff
@@ -0,0 +1,246 @@
+import org.springframework.kafka.annotation.EnableKafkaStreams;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.annotation.KafkaStreamsDefaultConfiguration;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaStreamsConfiguration;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.EmbeddedKafkaKraftBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+/**
+ * @author Soby Chacko
+ * @since 3.2.0
+ */
+@SpringJUnitConfig
+@DirtiesContext
+@EmbeddedKafka(partitions = 1,
+		topics = { "iqs-test-in", "iqs-test-out" })
+class KafkaStreamsInteractiveQueryServiceTests {
+
+	public static final String IQS_TEST_IN = "iqs-test-in";
+	public static final String IQS_TEST_OUT = "iqs-test-out";
+	public static final String STATE_STORE = "my-state-store";
+	public static final String NON_EXISTENT_STORE = "my-non-existent-store";
+
+	@Autowired
+	private EmbeddedKafkaKraftBroker embeddedKafka;
+
+	@Autowired
+	private StreamsBuilderFactoryBean streamsBuilderFactoryBean;
+
+	@Autowired
+	private KafkaTemplate<Integer, String> kafkaTemplate;
+
+	@Autowired
+	private KafkaStreamsInteractiveQueryService interactiveQueryService;
+
+	@Autowired
+	private CompletableFuture<ConsumerRecord<?, String>> resultFuture;
+
+	@Test
+	void retrieveQueryableStore() throws Exception {
+		this.kafkaTemplate.sendDefault(123, "123");
+		this.kafkaTemplate.flush();
+
+		ConsumerRecord<?, String> result = resultFuture.get(600, TimeUnit.SECONDS);
+		assertThat(result).isNotNull();
+
+		final ReadOnlyKeyValueStore<Object, Object> objectObjectReadOnlyKeyValueStore = this.interactiveQueryService
+				.retrieveQueryableStore(STATE_STORE,
+						QueryableStoreTypes.keyValueStore());
+
+		assertThat(objectObjectReadOnlyKeyValueStore.get(123)).isNotNull();
+		assertThat((Long) objectObjectReadOnlyKeyValueStore.get(123) >= 1L).isTrue();
```

Comment:
I don't think we need to call `get()` for the same value twice.
There is also `isGreaterThanOrEqualTo()` instead of that `isTrue()` for more precise reporting (if that).